starlark/eval.go

```diff
@@ -46,6 +48,21 @@ type Thread struct {
 	// See example_test.go for some example implementations of Load.
 	Load func(thread *Thread, module string) (StringDict, error)
 
+	// Steps is a count of abstract computation steps executed by
+	// this thread. It is incremented by the interpreter. If after
+	// being incremented, its value is zero, the interpreter calls
+	// thread.Cancel("too many steps").
+	//
+	// Steps may be used as a measure the approximate cost of
+	// Starlark execution, by computing the difference in its value
+	// before and after a computation. It may also be used to limit
+	// a computation to k steps by setting its initial value to
+	// 1<<64 - k.
```

Comment:
Consider adding a note that the number of steps taken by a given computation is an implementation detail and may change.